File: patterns/javascript-patterns/pattern113.js
// Pattern 113: ...

// *     *
// **   **
// *** ***
// *******
// *** ***
// **   **
// *     *

let n = 7;
let px = 1;

for (let i = 1; i <= n; i++) {
    let row = "";
    for (let j = 1; j <= n; j++) {
        if (j <= px || j >= n - px + 1) {
            row += "*";
        } else {
            row += " ";
        }
    }

    if (i <= n / 2) {
        px++;
    } else {
        px--;
    }
    console.log(row);
}

File: patterns/python-patterns/pattern113.py
# Pattern 113: ...

# *     *
# **   **
# *** ***
# *******
# *** ***
# **   **
# *     *

n = 7
px = 1

for i in range(1, n + 1):
    for j in range(1, n + 1):
        if j <= px or j >= n - px + 1:
            print("*", end="")
        else:
            print(" ", end="")

    if i <= n // 2:
        px += 1
    else:
        px -= 1

    print()
